src/community/ardrive_community_oracle.ts

```diff
@@ -0,0 +1,87 @@
+import { weightedRandom } from 'ardrive-core-js';
+import { ContractOracle } from './contract_oracle';
+import { CommunityOracle } from './community_oracle';
+import { ArweaveAddress, Winston } from '../wallet_new';
+import { ArDriveContractOracle } from './ardrive_contract_oracle';
+import Arweave from 'arweave';
+
+/**
+ * Minimum ArDrive community tip from the Community Improvement Proposal Doc:
+ * https://arweave.net/Yop13NrLwqlm36P_FDCdMaTBwSlj0sdNGAC4FqfRUgo
+ */
+export const minArDriveCommunityWinstonTip = 10_000_000;
+
+/**
+ * Oracle class responsible for determining the community tip
+ * and selecting the PST token holder for tip distribution
+ *
+ * TODO: Unit testing for important functions
+ */
+export class ArDriveCommunityOracle implements CommunityOracle {
+	constructor(readonly arweave: Arweave) {
+		this.contractOracle = new ArDriveContractOracle(arweave);
+	}
+
+	private readonly contractOracle: ContractOracle;
+
+	/**
+	 * Given an AR data cost, returns a calculated ArDrive community tip amount in AR
+	 *
+	 * TODO: Use big int library on Winston types
+	 */
+	async getCommunityWinstonTip(winstonCost: Winston): Promise<Winston> {
+		const communityTipValue = await this.contractOracle.getTipSettingFromContractSettings();
```

Comment:
Nit: `Value` isn't that useful for comprehension in this variable name. Consider `communityTipPercentage`. Also double check that it's not already a proportion vs. a percentage.